src/helpers/chain.js

```diff
@@ -0,0 +1,582 @@
+    i < envvars.SUBMIT_RETRIES && !submitted && !inSubmission;
+    i++
+  ) {
+    // Query round
+    let latestRound = await queryRound(feed);
+
+    /**
+     * If latestRound is greater than round being pushed or submission to the
+     * round is already made, abort
+     */
+    let latestRoundGreater = latestRound.roundId > round;
+    let submissionAlreadyMade =
+      latestRound.roundId === round && latestRound.submissionMade;
+    if (latestRoundGreater || submissionAlreadyMade) {
+      logger.info("Price failed to be submitted for old round: " + round);
+      return false;
+    }
+
+    logger.info("Submitting price for round " + round + " attempt " + (i + 1));
+
+    offer.id = Number(Date.now());
+
+    // Output action
+    let data = outputAction({
+      method: "executeOffer",
+      offer,
+    });
+
+    // Execute
+    await execSwingsetTransaction(
+      "wallet-action --allow-spend '" + JSON.stringify(data) + "'",
+      networkConfig,
+      from,
+      false,
+      keyring
+    );
+
+    // Update last submission time
+    await updateTable(
+      "jobs",
+      { last_submission_time: Date.now() / 1000 },
+      feed
+    );
+
+    // Sleep SEND_CHECK_INTERVAL seconds
+    await delay((Number(envvars.SEND_CHECK_INTERVAL) + 1) * 1000);
+
+    // Check submission for round
+    submitted = await checkSubmissionForRound(from, previousOffer, round);
+
+    // Check if in submission
+    inSubmission = await checkIfInSubmission(feed);
+  }
+
+  if (submitted) {
+    logger.info("Price submitted successfully for round " + round);
+  } else {
+    logger.error("Price failed to be submitted for round " + round);
+  }
+
+  return submitted;
+};
+
+/**
+ * Function to get offers and balances
+ * @param {Promise<import('@agoric/casting/src/follower-cosmjs').
+ * ValueFollower<T>>} follower offers and balances
+ * @param {string} oracle oracle address
+ * @returns {object} an object containing the offers and balances
+ * @returns {object[]} returns.offers Array of offers
+ * @returns {object[]} returns.balances Array of balances
+ */
+export const getOffersAndBalances = async (follower, oracle) => {
+    
+  let toReturn = {
+    offers: await getOffers(follower),
+    balances: [],
+  };
+
+  // Get current purses
+  let fromBoard = makeFromBoard();
+  const vstorage = makeVStorage({ fetch });
+  let current = await getCurrent(oracle, fromBoard, { vstorage });
+  for (let i = 0; i < current.purses.length; i++) {
+    let currentPurse = current.purses[i];
+    toReturn["balances"].push(currentPurse.balance);
+  }
+
+  return toReturn;
+};
+
+/**
+ * Function to get the amount in of a feed
+ * @param {string} feed feed name like ATOM-USD
+ * @returns {number} the amount in for the feed
+ */
+export const getAmountsIn = async (feed) => {
+  const capDataStr = await readVStorage(feed, false)
+
+  //parse the value
+  let capData = JSON.parse(JSON.parse(capDataStr).value);
+  capData = JSON.parse(capData.values[0]);
```

Comment:
Can't you JSON.parse the entire object and then access the elements in it?

Reply:
No because in CapData each object is a string and needs to be parsed